### PR TITLE
[6.0 🍒] Introduce VisionOS Platform

### PR DESCRIPTION
**Pull request by artemcm**
Cherry-pick of https://github.com/apple/swift/pull/72834
--------------------------------------
**Explanation**: This change introduces a new compilation target platform to the Swift compiler - visionOS.
- Changes to the compiler build infrastrucuture to support building compiler-adjacent artifacts and test suites for the new target.
- Addition of the new platform kind definition.
- Support for the new platform in language constructs such as compile-time availability annotations or runtime OS version queries.
- Utilities to read out Darwin platform SDK info containing platform mapping data.
- Utilities to support re-mapping availability annotations from iOS to visionOS (e.g. 'updateIntroducedPlatformForFallback', 'updateDeprecatedPlatformForFallback', 'updateObsoletedPlatformForFallback').
- Additional tests exercising platform-specific availability handling and availability re-mapping fallback code-path.
- Changes to existing test suite to accomodate the new platform.

**Risk**: Low, additional code-paths should have little effect on compilation for existing platforms.
**Testing**: Automated tests added to the compiler test suite. Existing tests adjusted for the new platform.
**Reviewed By**: @tshortli, @ktoso, @CodaFi, @nkcsgexi, @shahmishal

### CODE REVIEW

**Comment by artemcm** [comment]
@swift-ci test

File: lib/ClangImporter/ClangImporter.cpp [hunk 1]
```diff
@@ -609,6 +609,13 @@ importer::getNormalInvocationArguments(
     if (clangSupportsPragmaAttributeWithSwiftAttr())
       invocationArgStrs.push_back("-D__SWIFT_ATTR_SUPPORTS_SENDABLE_DECLS=1");
 
+    if (triple.isXROS()) {
+      // FIXME: This is a gnarly hack until some macros get adjusted in the SDK.
+      invocationArgStrs.insert(invocationArgStrs.end(), {
```
Comment (CodaFi) [inline lib/ClangImporter/ClangImporter.cpp:614]: Note to self: delete this.

**Review by CodaFi — APPROVED** [review]

**Review by nkcsgexi — APPROVED** [review]

**Comment by artemcm** [comment]
@swift-ci test

**Comment by artemcm** [comment]
@swift-ci test

**Comment by artemcm** [comment]
@swift-ci test macOS platform

**Pull request merged**